File: app/src/main/java/com/example/carlauncher/model/Gear.java
package com.example.carlauncher.model;

public enum Gear {
    P(0),
    R(1),
    N(2),
    D(3);

    private final int value;
    Gear(int value) { this.value = value; }
    public int getValue() { return value; }

    public static Gear fromString(String gearStr) {
        if (gearStr == null) return P;
        switch (gearStr.toUpperCase()) {
            case "R": return R;
            case "N": return N;
            case "D": return D;
            case "P":
            default: return P;
        }
    }
}

File: app/src/main/java/com/example/carlauncher/model/Gear.kt
package com.example.carlauncher.model

enum class Gear private constructor(val value: Int) {
    P(0),
    R(1),
    N(2),
    D(3);

    companion object {

        @JvmStatic
        fun fromString(gearStr: String?): Gear {
            if (gearStr == null) return P
            when (gearStr.uppercase(java.util.Locale.getDefault())) {
                "R" -> return R
                "N" -> return N
                "D" -> return D
                "P" -> return P
                else -> return P
            }
        }
    }
}
